fix: use supported node range for types

jrandolf-google opened the pull request:
Using the supported node range for types ensures Puppeteer compiles to the correct types.

Fixed: https://github.com/puppeteer/puppeteer/issues/10891

Lightning00Blade commented on packages/puppeteer/package.json:
```diff
@@ -30,9 +30,6 @@
     "type": "git",
     "url": "https://github.com/puppeteer/puppeteer/tree/main/packages/puppeteer"
   },
-  "engines": {
```
Why remove? Keep this it prints a warning about you using older version. There was a report about this sometime ago.

jrandolf committed:
fix: use supported node range for types

Using the supported node range for types ensures Puppeteer compiles to the correct types.
